lib/tissue-specific-score.js

```diff
@@ -88,3 +91,112 @@ export function getDataWithTissueScore(data, normalizedTissueSpecificScore) {
   }
   return [];
 }
+
+/**
+ * This function get filtered data using organ filters
+ * @param {Array} data all the data before filtering
+ * @param {Array} organFilters a list of organs for filtering the data
+ * @returns a list of filtered datasets
+ */
+export function getFilteredData(data, organFilters) {
+  let filteredData = data;
+  if (organFilters.length > 0) {
+    filteredData = filteredData.filter((dataset) => {
+      const organsForDataset = dataset.biosample_ontology?.organ_slims
+        ? dataset.biosample_ontology.organ_slims
+        : [];
+      return organsForDataset.some((organ) => organFilters.includes(organ));
+    });
+  }
+  return filteredData;
+}
+
+/**
+ * @param {Array} data used to generate organ facet
+ * @returns organ facet. Each organ is scored using its normalized tissue specific score. This score is used for coloring the body map.
+ */
+export function getOrganFacets(data, normalizedTissueSpecificScore) {
+  const organForData = [];
+  data.forEach((dataset) => {
+    const organSlim = dataset.biosample_ontology?.organ_slims
+      ? dataset.biosample_ontology.organ_slims
+      : [];
+    organSlim.forEach((organ) => {
+      if (!(organ in organForData)) {
+        organForData.push(organ);
+      }
+    });
+  });
+  const organFacet = {};
+  organForData.forEach((organ) => {
+    if (organ in normalizedTissueSpecificScore) {
+      organFacet[organ] = normalizedTissueSpecificScore[organ];
+    }
+  });
+  return organFacet;
+}
+
+/**
+ * @param {object} facets to get the score for the target organ
+ * @param {string} organ to fill color
+ * @returns the color in tailwind
+ */
+export function getFillColorTailwind(facets, organ) {
+  const fill = organ in facets ? TissueScoreTailwindColor[facets[organ]] : null;
+  return fill;
+}
+
+/**
+ * @param {object} facets to get the score for the target organ
+ * @param {string} organ to fill color
+ * @returns the color in hex
+ */
+export function getFillColorHex(facets, organ) {
+  const fill = organ in facets ? TissueScoreHexColor[facets[organ]] : "white";
+  return fill;
+}
```

Comment:
In the future, might be worth thinking about using custom Tailwind CSS classes instead of hex colors. Without dark-mode support, whether custom Tailwind CSS classes is easier or harder to maintain depends on the specific case and on your preferences.

Reply:
Is this something new? I didn't find how to use tailwind css classes for fill before. Thanks for the tips. I will create ticket to work on it.

Reply:
Custom Tailwind classes have been around probably since it started, and I’ve been using them since igvf-ui started. Maybe I shouldn’t say they can make things more difficult, but whether they actually help or not really depends on the situation. With dark mode support, they help simplify the React code a little or a lot.